### PR TITLE
Remove hack to get target of GetAccessor symbol

### LINKED ISSUE #27663: Can't rename property called `size`
**TypeScript Version:**  3.2.0--dev.20181009

**Search Terms:**
- rename
- tsserver

**Repo**

1. In the vscode [code base](
https://github.com/Microsoft/vscode) (sorry, couldn't find a smaller repo at the moment)
2. Open `src/vs/editor/contrib/codeAction/codeAction.ts`
3. After the imports, paste in:

```ts
interface Foo {
	size: number;
}
```

4. Try triggering a rename on `size`

**Bug**
Debug failure on TS Server:

```
Error: Debug Failure. False expression.
    at Object.getPropertySymbolFromBindingElement (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:92083:22)
    at forEachRelatedSymbol (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:98463:123)
    at getRelatedSymbol (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:98484:24)
    at getReferencesAtLocation (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:97979:37)
    at getReferencesInContainer (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:97942:21)
    at getReferencesInSourceFile (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:97929:24)
    at searchForName (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:97737:21)
    at getReferencedSymbolsForSymbol (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:97515:29)
    at Object.getReferencedSymbolsForNode (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:97421:57)
    at Object.findReferenceOrRenameEntries (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:97154:65)
    at getReferencesWorker (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:115440:41)
    at Object.findRenameLocations (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:115427:24)
    at /Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:121931:68
    at callbackProjectAndLocation (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:122016:13)
    at /Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:121990:24
    at forEachProjectInProjects (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:121974:17)
    at combineProjectOutputWorker (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:121988:13)
    at combineProjectOutputForRenameLocations (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:121929:13)
    at IOSession.Session.getRenameLocations (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:122918:33)
    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:122132:61)
    at /Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:123639:88
    at IOSession.Session.executeWithRequestId (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:123630:28)
    at IOSession.Session.executeCommand (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:123639:33)
    at IOSession.Session.onMessage (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:123661:35)
    at Interface.<anonymous> (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:124925:27)
    at emitOne (events.js:116:13)
    at Interface.emit (events.js:211:7)
    at Interface._onLine (readline.js:282:10)
    at Interface._normalWrite (readline.js:424:12)
    at Socket.ondata (readline.js:141:10)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at addChunk (_stream_readable.js:263:12)
    at readableAddChunk (_stream_readable.js:250:11)
    at Socket.Readable.push (_stream_readable.js:208:10)
    at Pipe.onread (net.js:594:20)
```

**Related Issues:**

### PR DESCRIPTION
**Pull request by andy-ms**
Fixes #27663
Essentially reverts #17483
Previously we had this special hack for following the target of the GetAccessor when it's transient.
Now we handle transient symbols appropriately by calling `checker.getRootSymbols(sym)` in `forEachRelatedSymbol`.

### CODE REVIEW

File: tests/cases/fourslash/findAllRefsDestructureGetter2.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/// <reference path="fourslash.ts" />
+
+// @noLib: true
+
+// @Filename: /a.ts
+////class C {
+////    get [|{| "isWriteAccess": true, "isDefinition": true |}m|](): number { return 0; }
+////}
+////const { [|{| "isWriteAccess": true, "isDefinition": true |}m|] } = new C();
+
```
Comment (sheetalkamat) [inline tests/cases/fourslash/findAllRefsDestructureGetter2.ts]: Can you add another property with set accessor only ?

File: src/services/utilities.ts [hunk 2]
```diff
@@ -1337,11 +1337,6 @@ namespace ts {
     export function getPropertySymbolFromBindingElement(checker: TypeChecker, bindingElement: ObjectBindingElementWithoutPropertyName) {
         const typeOfPattern = checker.getTypeAtLocation(bindingElement.parent);
         const propSymbol = typeOfPattern && checker.getPropertyOfType(typeOfPattern, bindingElement.name.text);
-        if (propSymbol && propSymbol.flags & SymbolFlags.Accessor) {
```
Comment (sheetalkamat) [inline src/services/utilities.ts]: This is also used in `updateImportsInOtherFiles` is there test for this?

**Comment by andy-ms** [comment]
@sheetalkamat Please re-review

File: src/services/utilities.ts (hunk 2 above)
Comment (andy-ms) [reply src/services/utilities.ts]: Fixed

File: tests/cases/fourslash/findAllRefsDestructureGetter2.ts (hunk 1 above)
Comment (andy-ms) [reply tests/cases/fourslash/findAllRefsDestructureGetter2.ts]: :+1:

File: tests/cases/fourslash/findAllRefsDestructureGetter2.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+/// <reference path="fourslash.ts" />
+
+// @noLib: true
+
+// @Filename: /a.ts
+////class C {
+////    get [|{| "isWriteAccess": true, "isDefinition": true |}g|](): number { return 0; }
```
Comment (sheetalkamat) [inline tests/cases/fourslash/findAllRefsDestructureGetter2.ts]: How is this writeAccess? Its a getter right?

Comment (andy-ms) [reply tests/cases/fourslash/findAllRefsDestructureGetter2.ts]: It's unintuitive but I think this is correct. `isWriteAccess` tells us when a symbol is given a value; in this case we're giving the getter a value. Just like a `const` declaration can be `isWriteAccess` even though the variable isn't mutable.

**Pull request merged**